doc-content/enterprise-only/openshift/hacep-clientcode-ref.adoc

```diff
@@ -0,0 +1,43 @@
+[id='hacep-clientcode-ref']
+= Requirements for client code
+
+When developing client code for high-availability CEP, you must follow certain additional requirements.
+
+[float]
+== kie-remote API
+
+The code must use the kie-remote API and not the kie API. The kie-remote API is specified in the `org.kie:kie-remote` Maven artifact. You can find the source code in the `kie-remote` Maven module.
+
+[float]
+== Explicit timestamps
+
+The engine needs to determine the sequence in which events happen. For this reason every event must have an associated timestamp assigned to it. In a high-availability environment, make this timestamp a property of the Java bean modelling
```

Comment:
"The {DECISION_ENGINE} needs"

"For this reason, every"

"property of the JavaBean"  (it's a trademarked word, and I've mass-updated the project accordingly when I worked on the decision engine doc)

Also suggest "JavaBean that models the event"

"class with the `@Timestamp` ..."